lib/puppet/provider/package/dnfmodule.rb

```diff
@@ -0,0 +1,88 @@
+# dnfmodule - A puppet package provider for DNF modules
+#
+# Installing a module:
+#  package { 'postgresql':
+#   provider => 'dnfmodule',
+#   ensure   => '9.6',  # install a specific stream
+#   flavor   => 'client',  # install a specific profile
+# }
+
+
+require 'puppet/provider/package'
+
+Puppet::Type.type(:package).provide :dnfmodule, :parent => :dnf do
+
+  has_feature :installable, :uninstallable, :versionable
+  #has_feature :upgradeable
+  # it's not (yet) feasible to make this upgradeable since module streams don't
+  # always have matching version types (i.e. idm has streams DL1 and client,
+  # other modules have semver streams, others have string streams... we cannot
+  # programatically determine a latest version for ensure => 'latest'
+
+  commands dnf: '/usr/bin/dnf'
```

Comment:
Can't use keyword syntax in 5.5.x due to needing to support ruby 1.9/jruby 1.7

Reply:
I changed this, but this syntax works for me on ruby 1.9:

```
irb(main):001:0> { a: 'b', b: :c, d: 1 }
=> {:a=>"b", :b=>:c, :d=>1}
irb(main):002:0> RUBY_VERSION
=> "1.9.3"
```

Reply:
Sorry I was conflating keyword arguments with the shorter `key: value` notation. Looks good.